packages/cli/src/gemini.tsx

```diff
@@ -102,16 +102,49 @@ function getNodeMemoryArgs(config: Config): string[] {
 }
 
 async function relaunchWithAdditionalArgs(additionalArgs: string[]) {
-  const nodeArgs = [...additionalArgs, ...process.argv.slice(1)];
-  const newEnv = { ...process.env, GEMINI_CLI_NO_RELAUNCH: 'true' };
+  // If stdin is in raw mode, we need to temporarily disable it so that the
+  // child process can take control of it.
+  const wasRaw = process.stdin.isRaw;
+  if (wasRaw) {
+    process.stdin.setRawMode(false);
+  }
 
-  const child = spawn(process.execPath, nodeArgs, {
-    stdio: 'inherit',
-    env: newEnv,
-  });
+  // The parent process should not be reading from stdin while the child is running.
+  process.stdin.pause();
 
-  await new Promise((resolve) => child.on('close', resolve));
-  process.exit(0);
+  try {
+    const nodeArgs = [...additionalArgs, ...process.argv.slice(1)];
+    const newEnv = { ...process.env, GEMINI_CLI_NO_RELAUNCH: 'true' };
+
+    const child = spawn(process.execPath, nodeArgs, {
+      stdio: 'inherit',
+      env: newEnv,
+      detached: true,
```

Comment:
is this change to detached mode required?

Reply:
Reverted since its not making a diff on my system.